crunchers/cruncher.h

```diff
@@ -29,58 +29,169 @@
+  GenericCruncherImpl(uint16_t port, const std::string& route, ARGS&&... args)
+      : IMPL(std::forward<ARGS>(args)...),
+        port_(port),
+        mmq_subscriber_([this](mmq_message_t&& message, idxts_t) { message->Handle(*this); }),
+        mmq_(mmq_subscriber_, 1024 * 1024) {
+    scoped_http_routes_ += HTTP(port).Register(route + "/healthz", [](Request r) { r("OK\n"); }) +
+                           HTTP(port).Register(route + "/data",
+                                               [this](Request r) {
+                                                 mmq_.Publish(std::make_unique<RequestMessage>(std::move(r)));
+                                               });
+  }
+  virtual ~GenericCruncherImpl() = default;
+
+  void Join() { HTTP(port_).Join(); }
+
+  EntryResponse operator()(const event_t& event, idxts_t current, idxts_t) {
+    mmq_.Publish(std::make_unique<EventMessage>(event, current));
+    return EntryResponse::More;
+  }
+
+  EntryResponse operator()(event_t&& event, idxts_t current, idxts_t) {
+    mmq_.Publish(std::make_unique<EventMessage>(std::move(event), current));
     return EntryResponse::More;
   }
 
   EntryResponse EntryResponseIfNoMorePassTypeFilter() const { return EntryResponse::More; }
   TerminationResponse Terminate() const { return TerminationResponse::Terminate; }
 
  private:
-  subscriber_callback_t actual_subscriber_;
+  const uint16_t port_;
+  HTTPRoutesScope scoped_http_routes_;
+  IntermediateSubscriber<mmq_message_t> mmq_subscriber_;
+  mmq_t mmq_;
 };
 
-template <typename ENTRY>
-using IntermediateSubscriber = current::ss::StreamSubscriber<IntermediateSubscriberImpl<ENTRY>, ENTRY>;
+template <typename IMPL>
+using StreamCruncher = current::ss::StreamSubscriber<GenericCruncherImpl<IMPL>, typename IMPL::event_t>;
+
+CURRENT_STRUCT_T(CruncherResponse) {
+  CURRENT_FIELD(comment, std::string);
+  CURRENT_FIELD(timestamp, std::chrono::microseconds, std::chrono::microseconds(0));
+  CURRENT_FIELD(value, T);
+};
+
+template <typename CRUNCHER>
+class MultiCruncher {
+ public:
+  using cruncher_t = CRUNCHER;
+  using event_t = typename cruncher_t::event_t;
+  using value_t = typename cruncher_t::value_t;
+  using value_list_t = std::vector<value_t>;
+  using response_t = CruncherResponse<value_list_t>;
+  using response_short_t = CruncherResponse<value_t>;
+
+  template <typename ARG>
+  MultiCruncher(const std::vector<ARG>& args)
+      : last_event_us_(std::chrono::microseconds(0)) {
+    crunchers_.reserve(args.size());
+    for (const auto& arg : args) {
+      crunchers_.push_back(std::make_unique<cruncher_t>(arg));
+    }
+  }
+  virtual ~MultiCruncher() = default;
+
+  void OnEvent(event_t&& event, idxts_t idxts) {
+    last_event_us_ = idxts.us;
+    for (auto& cruncher : crunchers_) {
+      cruncher->OnEvent(std::move(event), idxts);
+    }
+  }
+
+  void OnRequest(Request&& r) {
+    const std::string comment =
+        Printf("Last processed event was %llu minutes ago",
```

Comment:
How about two fields, one `std::chrono::microseconds`, and one `std::string`, set via `TimeIntervalAsHumanReadableString` from `Bricks/strings/time.h` ?